### PR TITLE
Implement generic pattern

### PR DESCRIPTION
**Pull request by spbnick**
This implements generic test condition patterns, which are to replace the use of `match` and `dont_match`. Required by kpet-db MR 251, which also contains plenty of new pattern usage examples and some documentation.

### CODE REVIEW

**Comment by spbnick** [comment]
@cki-bot, could you please test this?

File: kpet/data.py [hunk 1]
```diff
@@ -95,6 +96,143 @@ def normalize(arg):
...
+    # Target field qualifiers
+    qualifiers = {"trees", "arches", "sets", "sources", "location_types"}
+
+    """An execution target pattern"""
+    def __init__(self, data):
+        """
+        Initialize an execution pattern.
+
+        Args:
+            positive:   True if the pattern is positive, False if negative.
```
Comment (veruu) [inline kpet/data.py]: This is not on the `__init__` line nor used in the code

Comment (spbnick) [reply kpet/data.py]: Argh, an overlooked copy-paste. Thank you :)

**Comment by spbnick** [comment]
There were no kpet-related failures in cki-bot results. The issue found by @veruu is fixed. Merging.

**Pull request merged**